torch/testing/_internal/opinfo/definitions/signal.py

```diff
@@ -0,0 +1,269 @@
+import random
+import unittest
+
+from itertools import product
+from typing import List
+
+import torch
+from torch.testing._internal.common_dtype import all_types_and
+from torch.testing._internal.common_utils import TEST_SCIPY
+from torch.testing._internal.opinfo.core import (
+    DecorateInfo,
+    ErrorInput,
+    OpInfo,
+    SampleInput,
+)
+
+if TEST_SCIPY:
+    import scipy.signal
+
+
+def sample_inputs_window(op_info, device, dtype, requires_grad, *, kws=({},), **kwargs):
+    _kwargs = dict(
+        kwargs, **{"device": device, "dtype": dtype, "requires_grad": requires_grad}
+    )
+    sizes = [0, 1, 2, 5, 10, 50, 100, 1024, 2048]
+    for size, periodic, k in product(sizes, (True, False), kws):
+        yield SampleInput(size, periodic=periodic, **k, **_kwargs)
+
+
+def sample_inputs_gaussian_window(op_info, device, dtype, requires_grad, **kwargs):
+    kws = [{"std": random.uniform(0, 3)} for _ in range(50)]
```

Comment:
5 is too many. We usually just test one value for each input. These samples are run with *many* tests in *every* CI run in *every* PR. We don't want to generate too many samples, we just want to generate enough samples to cover all the relevant cases.

Reply:
If you want to generate extra inputs, you can look into generating `reference_inputs` (you can read about them in `test_ops.py`.  Now, I don't think that's necessary in this case.

Also, think that these args will be run in all the CI boxes. Chances are tha the `std` parameter is different for each CI box. That will give you enough parameters to be sure this is tested correctly :)

Reply:
Oh, alright. I get it now. I'll work on it tonight.

Reply:
I updated this section. I'll tag you in another comment where the new code lives.